crates/biome_js_analyze/src/lint/nursery/use_exports_last.rs

```diff
@@ -0,0 +1,80 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsModuleItem, JsModuleItemList, JsSyntaxNode};
+use biome_rowan::AstNode;
+use biome_rowan::AstNodeList;
+
+declare_lint_rule! {
+    /// Require that all exports are declared after all non-export statements.
+    ///
+    /// Enforces that export statements are placed at the end of the module, after all other statements.
+    ///
+    /// Source: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/exports-last.md
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// export const a = 1;
+    /// const b = 2;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const a = 1;
+    /// export const b = 2;
+    /// ```
+    ///
+    /// ```js
+    /// const a = 1;
+    /// export { a };
+    /// ```
+    ///
+    pub UseExportsLast {
+        version: "next",
+        name: "useExportsLast",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintImport("exports-last")],
+    }
+}
+
+impl Rule for UseExportsLast {
+    type Query = Ast<JsModuleItemList>;
+    type State = JsSyntaxNode;
```

Comment:
I think `TextRange` is enough
```suggestion
    type State = TextRange;
```